common/dev/pmic.c

```diff
@@ -29,7 +29,7 @@ LOG_MODULE_REGISTER(pmic);
 uint8_t *compose_memory_write_read_req(uint8_t smbus_identifier, uint8_t smbus_address,
 				       uint32_t addr_value, uint8_t *write_data, uint8_t write_len)
 {
-	if ((write_len != 0) && (write_data == NULL)) {
+	if ((write_data == NULL) && (write_len != 0)) {
```

Comment:
This could be changed to just a CHECK_NULL_ARG_WITH_RETURN(write_data, NULL); for the write_data == NULL check.  The second check for write_len !=0 I believe is not correct.  There should be a check for write_len == 0 with a LOG_ERR and a return as the function is using the write_len to determine how many bytes to copy in the memcpy below.

Reply:
I think the code as written (either with this change or without it) is correct.  This allows someone to formulate a header without data attached to it.  If the data is null, the length also has to be 0.

I've added a guard around the memcpy for clarity as well.